root_name: "ur5ebase_link"
tip_name: "ur5ewrist_3_link"
# [position (x, y, z)  orientation (x, y, z, w)]
desired_pose:  [0.0,0.40,0.65, 0.707, -0.707, 0.0, 0.0]
Ka: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
Kv: [10, 10, 10, 10, 10, 10]
Kp: [100, 100, 100, 100, 100, 100]


# J^T ============================================================================================
# PID-VIC performance
#Wf: 0
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [100, 100, 100, 1, 1, 1]
#stiffness_coupling: [500, 500, 500, 25, 25, 25] 
#Kf: [0.0, 0.0, 0.0, 0., 0., 0.]

# UDE-vic performance                 # 0 ss error
#Wf: 50
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [0.5, 0.5, 0.5, 0.005, 0.005, 0.005]
#stiffness_coupling: [0.2, 0.2, 0.2, 0.005, 0.005, 0.005]
#Kf: [0, 0., 0, 0, 0, 0]

# ================= performance ============================== 
# normal
# Wf: 0
Wf: 0
mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
damping_arm: [20, 20, 20, 1, 1, 1]
stiffness_coupling: [200, 200, 200, 20, 20, 20] 
Kf: [0.0, -5.0, 0.0, 0., 0., 0.]

# mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
# damping_arm: [0.67, 0.67, 0.67, 0.033, 0.033, 0.033]
# stiffness_coupling: [6.67, 6.67, 6.67 ,0.67, 0.67, 0.67]
# Kf: [0, 0., 0, 0, 0, 0]
# to 0.02


#Wf: 60
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [0.2, 0.2, 0.2, 0.001, 0.001, 0.001]
#stiffness_coupling: [0.4, 0.4, 0.4, 0.01, 0.01, 0.01]
#Kf: [0, 0., 0, 0, 0, 0]
# ===============================================================================================
## low pass filter and impedance parameters
# PID-VIC performance
#Wf: 0.
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [100, 100, 100, 1, 1, 1]
#stiffness_coupling: [500, 500, 500, 25, 25, 25] 

#damping_arm: [10, 10, 10, 0.1, 0.1, 0.1]
#stiffness_coupling: [50, 50, 50, 5, 0.5, 5] 


# UDE50-vic performance                 # 0 ss error
#Wf: 50
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [0.5, 0.5, 0.5, 0.005, 0.005, 0.005]
#stiffness_coupling: [0.2, 0.2, 0.2, 0.005, 0.005, 0.005] 

#damping_arm: [0.5, 0.5, 0.5, 0.01, 0.01, 0.01]
#stiffness_coupling: [0.2, 0.2, 0.2, 0.015, 0.015, 0.015] 

#damping_arm: [0.5, 0.5, 0.5, 0.01, 0.01, 0.01]
#stiffness_coupling: [0.2, 0.2, 0.2, 0.02, 0.02, 0.02] 


#damping_arm: [0.1, 0.1, 0.1, 0.01, 0.001, 0.01]
#stiffness_coupling: [10, 10, 10, 0.5, 0.05, 0.5] 


# UDE25-vic performance                 # 0.075m 0.03m ss error
#Wf: 25
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [0.2, 0.2, 0.2, 0.02, 0.002, 0.02]
#stiffness_coupling: [20, 20, 20, 1.0, 0.1, 1.0] 


# UDE50-p-10                            # 0.18m 0.05m ss error
# UDE50-p-15 vic                        # 0.08m 0.042m ss error      
# UDE50-p-20 vic                        # 0.07m 0.035m ss error      

# greater p means smaller sse, but i seems have no effects. 
# =========================================== p25 exp =====================================================
#Wf: 50
#mass_arm: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
#damping_arm: [0.1, 0.1, 0.1, 0.01, 0.001, 0.01]
#stiffness_coupling: [10, 10, 10, 0.5, 0.05, 0.5] 


# 0.055m 0.026m ss error
#damping_arm: [0.1, 0.1, 0.1, 0.01, 0.001, 0.01]
#stiffness_coupling: [25, 25, 25, 0.5, 0.05, 0.5] 

# 0.05m 0.025m ss error
#damping_arm: [0.25, 0.25, 0.25, 0.01, 0.001, 0.01]
#stiffness_coupling: [25, 25, 25, 0.5, 0.05, 0.5] 

#damping_arm: [0.3, 0.3, 0.3, 0.01, 0.001, 0.01]
#stiffness_coupling: [30, 30, 30, 1.5, 0.15, 1.5]

# 0.05m 0.025m ss error this one good
#damping_arm: [0.25, 0.25, 0.25, 0.01, 0.001, 0.01]
#stiffness_coupling: [25, 25, 25, 1.5, 0.15, 1.5] 

#damping_arm: [0.45, 0.45, 0.45, 0.02, 0.002, 0.02]
#stiffness_coupling: [50, 50, 50, 1.5, 0.15, 1.5] 

#damping_arm: [0.55, 0.55, 0.55, 0.01, 0.001, 0.01]
#stiffness_coupling: [50, 50, 50, 0.5, 0.05, 0.5] 

# 0.00225099 0.00023878 0.00108529 50s 10 0.1
# 0.00208197 0.00023832 0.0010228 50s 10 0
#             unstable   10   0.5

# 0.05m 0.025m ss error but more shakey
#damping_arm: [0.25, 0.25, 0.25, 0.03, 0.003, 0.03]
#stiffness_coupling: [25, 25, 25, 2.5, 0.25, 2.5] 
# =================================================================

#Kf: [100, 100, 100, 100, 100, 100]
#Kf: [10, 10, 10, 10, 10, 10]
#Kf: [0, 0, 10, 0, 0, 0]
#Kf: [5, 5, 5, 5, 5, 5]
#Kf: [1, 1, 1, 1, 1, 1]

#Kf: [0, 0, 0, 0, 0, 0]


publish_rate: 1000

Car_K_v: 20
Car_K_w: 15
